web/new_alt_text_manager.js

```diff
@@ -274,6 +288,34 @@ class NewAltTextManager {
     this.#textarea.value = altText;
   }
 
+  #setProgress() {
```

Comment:
This method is unfortunately neither particularly readable nor easily understandable, at least to me. Please see if the code can be re-factored somehow.